### DISCUSSION TITLE
Interesting Observation

### DISCUSSION THREAD

**Discussion by N6RFM** [daniestevez/galileo-osnma#15] [General]
Hi Dani, 

Scott and I have each been playing with some additional galmon features.  This includes being able to stream Galileo satellite status information to a local web page, and having the osnma-longan-nano-client run at the same time.  In several instances, we have noticed that while the portal information suggests for a specific satellite that OSNMA is OFF, we are still seeing what appear to be valid processing results for the satellite in question.  See attached examples for E18. The OSNMA ON/OFF status is the second to last column.  This result seems odd?  Is it possible that OSNMA status is not relevant for OSNMA information reception and subsequent decoding?  Or, maybe something else is going on?
![Selection_218](https://user-images.githubusercontent.com/36707700/162597608-fe8805a8-1fcb-445d-93c2-11b01e72c6c8.png)
![Selection_217](https://user-images.githubusercontent.com/36707700/162597609-997a0e1d-e849-41b8-8ea0-5b1da678ee85.png)

Thanks,
Bob

**Comment by daniestevez** [comment]
I think the OSNMA status you can see on the Galmon web page is accurate. What happens is that the Galileo satellites send signatures for the data corresponding to all of the satellites in the constellation. So when you see `E18 TOW 597090` that means you have received the navigation data from E18 and a signature for that data. However, the signature need not come from E18. Another satellite could have transmitted it. The log that osnma-longan-nano shows is very basic and doesn't give these details, but with galmon-osnma (the PC application) you can see where the signatures come from. For instance,
```
E27 InavCed at Gst { wn: 1176, tow: 121050 } tag1 correct (auth by E19)
```
means that it has successfully checked a signature for the navigation data from E27, but the signature was transmitted by E19.

**Comment by K4KDR** [comment]
That's very helpful - thank you!

But on the subject of the output from the **nano**, is the following output a packet from E21 that whose signature is authenticated via E04, or a packet from E04 whose signature is authenticated via E21?  Or something else?

```
[2022-04-10 13:22:24] E21 WN 1181 TOW 48149 OSNMA
[2022-04-10 13:22:24] AUTH ADKD=4 NONE
[2022-04-10 13:22:24] AUTH ADKD=0 E04 TOW 48060
```

-Scott

**Comment by daniestevez** [comment]
Good question. That's actually an OSNMA data packet transmitted by E21 that in turn triggered the authentication of data for E04. However (and this a tricky thing about how OSNMA works), the signature (tag) that has been used to check the data for E04 might have been transmitted by another satellite. What happens it that to check tags, a TESLA key that is transmitted 30 seconds later than the tags is needed. So the tags received from all the satellites are stored in RAM until they can be used. At this point in time, the OSNMA packet from E21 allows us to grab the new TESLA key (all the satellites transmit the same TESLA key at the same time, so nothing special here about E21) and go through all the tags we stored 30 seconds ago (osnma-longan-nano doesn't print much detail about which satellites these came from) and authenticate the data for E04. So we know that the TESLA key was obtained from E21, but we don't know from which satellite/satellites the tags came from (actually checking two tags is needed to consider that the data is authenticated).

**Comment by N6RFM** [comment]
Thanks Dani and Scott for informative explanations!
-Bob
